src/ProgramRepresentation/ProgramFunction.cpp

```diff
@@ -0,0 +1,57 @@
+#include "ProgramRepresentation/ProgramFunction.h"
+#include "Debug.h"
+#include "ProgramRepresentation/ProgramVariable.h"
+
+ProgramFunction::ProgramFunction(std::string name) { this->name = name; }
+
+ProgramFunction::ProgramFunction() {}
+
+void ProgramFunction::addProgramPoint(ProgramPoint programPoint) {
+  this->programPoints.push_back(programPoint);
+}
+
+std::list<ProgramPoint> ProgramFunction::getProgramPoints() {
+  return this->programPoints;
+}
+
+ProgramPoint *ProgramFunction::getProgramPointRef(std::string name,
+                                                  bool addNewIfNotFound) {
+  for (ProgramPoint &programPoint : this->programPoints) {
+    if (programPoint.getName() == name) {
+      return &programPoint;
+    }
+  }
+
+  if (addNewIfNotFound) {
+    ProgramPoint newProgramPoint = ProgramPoint(name);
+    this->addProgramPoint(newProgramPoint);
+    return &this->programPoints.back();
+  }
+
+  return new ProgramPoint();
```

Comment:
I now realize that it should error instead because it returning a `new ProgramPoint` really shouldn't be happening. For now, I'll add a todo saying that it should error instead. Also I'll update the docs

Reply:
Why add a TODO and not just have it error now? I don't think it would be that hard to change this to throw some kind of exception if this case is really expected never to occur.